### PR TITLE
Functional documentation (variants, visibilty and exclude from sale)

### PR DESCRIPTION
**Pull request by PetraPiprekova**
| Q             | A
| ------------- | ---
|Description, reason for the PR| ...
|New feature| No 
|BC breaks| No 
|Fixes issues| ...
|Standards and tests pass| No
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/behavior-of-variants.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Behavior of variants
+
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/behavior-of-variants.md]: line deleted

File: docs/behavior-of-variants.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+Variants are used for products that are suitable to associate. These are products that vary for example by size, pattern, color, etc. A typical example of using variants can be clothing where one model of t-shirt differs only in size.
+The use of variants shows the specific behavior described below.
+
+## Dictionary of terms
+
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/behavior-of-variants.md]: line deleted

File: docs/index.md [hunk 3]
```diff
@@ -37,6 +37,11 @@
 * [Adding Ajax Load More Button into Pagination](./cookbook/adding-ajax-load-more-button-into-pagination.md)
 * [Implementing a basic data import](./cookbook/basic-data-import.md)
 
+## Functional documentation
+* [Behavior of variants](./behavior-of-variants.md)
+* [Visibility and Exclude from sale](./visibility-and-exclude-from-sale.md)
+
+
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/index.md]: line deleted

File: docs/visibility-and-exclude-from-sale.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Visibility of products
+
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/visibility-and-exclude-from-sale.md]: line deleted

File: docs/visibility-and-exclude-from-sale.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+- the product can be viewed only via URL
+- the product is not displayed on the front-end of the product listing or in the search results
+- the product cannot be added to cart and customer is informed (by the text) about the end of the sale of the product instead of the purchase button
+
+
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/visibility-and-exclude-from-sale.md]: deleted

File: docs/visibility-and-exclude-from-sale.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+# Exclude from sale
+
+The product which is excluded from sale can be viewed via URL but can not be added to cart. The product which is discounted from the sale is not reachable within the eshopp. This means that product is not displayed in product overview, search results, filtering and is not generated to XML feeds. Excluded from the sale is used if we do not want to see a 404 error page hitting the product URL.
+
+The product is removed from the sale if any of the conditions are set in the product administration, in particular:
+- **Exclude from sale -> Yes**
+- **Action after sellout -> Exclude from sale**
+
+## Behavior of product that is excluded from the sale:
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/visibility-and-exclude-from-sale.md]: This paragraph describes the settings in administration. I would like to keep it.

File: docs/behavior-of-variants.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+2. In the form for creating a variant is necessary to choose from existing products main product first and then its variants.
+3. The “create” button confirms the creation of the product with variants.
+4. A new card is copied from the main product, which is the main product for the selected variants.
+5. The original main product is also added to the list of variants and can be purchased.
+6. So we have created the main product, which is abstract and specific variants can be assigned to that product.
+
+## Behavior of variants
+
+- ### Administration - main product:
+  - there is the icon "H" for the main product on the products overview
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/behavior-of-variants.md]: Yes :) David made new PR for this issue.

File: docs/visibility-and-exclude-from-sale.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (boris-brtan) [reply docs/visibility-and-exclude-from-sale.md]: PO says @PetraPiprekova is right

File: docs/visibility-and-exclude-from-sale.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
+# Visibility of products
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply docs/visibility-and-exclude-from-sale.md]: headline renamed

File: docs/visibility-and-exclude-from-sale.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply docs/visibility-and-exclude-from-sale.md]: desc added

File: docs/visibility-and-exclude-from-sale.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+If the product is not visible on the eshop, then you will see the 404 error page when you will enter its URL. The invisible product is not reachable within the eshop at all. This actually means the product is not displayed in product overview, search results, filtering and is not generated to XML feeds.
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply docs/visibility-and-exclude-from-sale.md]: removed the paragraph

File: docs/visibility-and-exclude-from-sale.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+**Visible product** - is displayed on the Front-end of the e-shop in products overview, search results, XML feeds etc.
+
+**Invisible product** - is not displayed on the Front-end of the e-shop, and page with 404 error is displayed hitting product URL.
+
+**Product exclude from sale** - is not displayed on Front-end of the e-shop but the product details are displayed using URL, however without the option to add product to cart.
+
+The product is not visible on the Front-end of the eshop in cases described below:
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply docs/visibility-and-exclude-from-sale.md]: added as suggested

**Pull request merged**